api/authentication/views.py

```diff
@@ -122,25 +121,25 @@ def delete(self, request, *args, **kwargs):
 class TeacherAPIView(ListAPIView, ListModelMixin, DestroyAPIView):
     lookup_field = 'id'
     serializer_class = UserSerializer
-    queryset = User.objects.all()
+    queryset = User.objects.none()
     permission_classes = [AllowAny,]
 
-    @permissions(["IsStaffUser", "IsUser"])
+    @permissions(["IsStaffUser", "IsUser"],"kwargs")
     def get(self, request, *args, **kwargs):
-        number = args[0].get('id')
```

Comment:
In decorator you're using kwargs, so args cause error. Need to be fixed. You have to use only kwargs variable